images/virtualization-artifact/pkg/controller/vmmac/internal/service/mac_address_service.go

```diff
@@ -0,0 +1,255 @@
+/*
+Copyright 2025 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package service
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/rand"
+	"regexp"
+	"strings"
+	"time"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/deckhouse/virtualization-controller/pkg/common/annotations"
+	"github.com/deckhouse/virtualization-controller/pkg/common/mac"
+	"github.com/deckhouse/virtualization-controller/pkg/common/object"
+	"github.com/deckhouse/virtualization-controller/pkg/logger"
+	"github.com/deckhouse/virtualization/api/client/kubeclient"
+	virtv2 "github.com/deckhouse/virtualization/api/core/v1alpha2"
+)
+
+const MaxCount int = 16777216
+
+type MACAddressService struct {
+	oui        string
+	client     client.Client
+	virtClient kubeclient.Client
+}
+
+func NewMACAddressService(
+	clusterUUID string,
+	client client.Client,
+	virtClient kubeclient.Client,
+) *MACAddressService {
+	oui := generateOUI(clusterUUID)
+
+	oui, err := formatOUI(oui)
+	if err != nil {
+		return nil
+	}
+
+	return &MACAddressService{
+		oui:        oui,
+		client:     client,
+		virtClient: virtClient,
+	}
+}
+
+func generateOUI(clusterUID string) string {
```

Comment:
A short description of transformation done here will be helpful for future us =)